### ISSUE TITLE
Release for GHC 8.6

### ISSUE THREAD

**Issue by bgamari** [haskell/cabal#5381]
It would be great to have GHC's core libraries sorted out by mid-July at the latest (by the end of June would be better).

**Comment by 23Skidoo** [comment]
I'll create a release branch next week.

**Comment by bgamari** [comment]
Thanks @23Skidoo!

**Comment by quasicomputational** [comment]
Is the next version going to be 2.4? The glob stuff is gated on 3.0, so I'll change that (+ tests, docs, etc) if it is.

**Comment by 23Skidoo** [comment]
We decided to do a 2.4, yep.

**Comment by quasicomputational** [comment]
I've got a bunch of PRs which are each individually smallish that I'd really like to see land before the release as well. I don't think anything is blocking these except that I'd like a second set of eyes on them before I go ahead and merge. I've put them all in the 2.4 milestone, if anyone would like a look.

**Comment by hvr** [comment]
@phadej btw, the reason to wait for that PR with all those carefully crafted logical commits is to avoid cherry-picking dozens of commits manually between branches... :-)

**Comment by phadej** [comment]
@hvr, yeah whatever is simpler&easier for Mikhail

**Comment by tom-bop** [comment]
Any chance `new-install` will make it in?

**Comment by typedrat** [comment]
I just need someone to look over the code and ideally try it on another platform or two. Seems to work at least for me, and I feel like it could make it in.

**Comment by int-index** [comment]
Let's not forget to add new extensions, `QuantifiedConstraints` and `StarIsType`.

**Comment by quasicomputational** [comment]
I've put up #5419 for those two new extensions.

**Comment by sboosali** [comment]
@typedrat I can test it on Windows and Ubuntu. Send me the branch, and tell me if I should do anything beyond running the test sweet and installing a random package. (Like, build a package under GHC 8.6 too? Or is there an integration tests or something that covers that.)

**Comment by bgamari** [comment]
Ping.

**Comment by bgamari** [comment]
Hello? I will be pushing out the 8.6.1 beta within the next week; I would strongly prefer not to have to do any further submodule bumps (and certainly no major version bumps) after this. Really it would be nice to have a more-or-less ready branch by early-to-mid next week. If this isn't possible perhaps then we could instead just cut a release from the 2.2 branch.

**Comment by 23Skidoo** [comment]
`2.4` branch created. Sorry for the delay.

**Comment by bgamari** [comment]
Thanks @23Skidoo!

**Comment by bgamari** [comment]
How is the final release looking?

**Comment by sboosali** [comment]
(btw, for a lay person to test this out, should i just clone the branch and
use it to build a project? do i have to clean out an old dist-newstyle, or
is the Cabal version part of the package hash? i have something small with
signatures, so maybe that's relevant.)

**Comment by bgamari** [comment]
Ping.

**Comment by 23Skidoo** [comment]
@bgamari I think we can do a lib:Cabal release this week and a cabal-install one soon after.

**Comment by 23Skidoo** [comment]
@sboosali It should work out of the box.

**Comment by bgamari** [comment]
@23Skidoo, right, let me know as soon as you have a commit.

**Comment by 23Skidoo** [comment]
@bgamari `Cabal-2.4.0.0` [tagged](https://github.com/haskell/cabal/releases/tag/Cabal-v2.4.0.0) and [released on Hackage](https://hackage.haskell.org/package/Cabal-2.4.0.0).

**Comment by bgamari** [comment]
Thanks @23Skidoo!

**Comment by 23Skidoo** [comment]
@bgamari I've released `Cabal-2.4.0.1` now as well, which has one important bugfix (#5569); would be nice if 8.6.1 included that version instead.
